base/abstractarray.jl

```diff
@@ -1207,6 +1207,7 @@ function copymutable(a::AbstractArray)
     copyto!(similar(a), a)
 end
 copymutable(itr) = collect(itr)
+copymutable(a::Array) = copy(a)
```

Comment:
Why not improving `copyto!(::Array, ::Array)` instead?

Reply:
I don't see how `copyto!(::Array, ::Array)` could ever be made as efficient as `copy(::Array)`. In the former case, the implementation must check the bounds of the destination array and also must check for aliasing. That's not needed in the latter.

That's not to say `copyto!(::Array, ::Array)` couldn't be made faster (I don't know), it's just to say that it seems to me that `copy(::Array)` is the most efficient choice here.